addons/a11y/src/components/A11YPanel.tsx

```diff
@@ -49,228 +48,133 @@ const Centered = styled.span<{}>({
-        }, 900);
+type Status = 'initial' | 'manual' | 'running' | 'error' | 'ran' | 'ready';
+
+export const A11YPanel: React.FC = () => {
+  const [status, setStatus] = React.useState<Status>('initial');
+  const [error, setError] = React.useState<unknown>(undefined);
+  const { setResults, results } = useA11yContext();
+  const { passes, incomplete, violations } = results;
+  const { storyId } = useStorybookState();
+  const { manual } = useParameter<Pick<A11yParameters, 'manual'>>('a11y', {
+    manual: false,
+  });
+
+  React.useEffect(() => {
+    setStatus(manual ? 'manual' : 'initial');
+  }, [manual]);
+
+  const handleResult = (axeResults: AxeResults) => {
+    setStatus('ran');
+    setResults(axeResults);
+
+    setTimeout(() => {
+      if (status === 'ran') {
+        setStatus('ready');
       }
-    );
-  };
-
-  onError = (error: unknown) => {
-    this.setState({
-      status: 'error',
-      error,
-    });
+    }, 900);
   };
 
-  onManual = (manual: boolean) => {
-    if (manual) {
-      this.setState({
-        status: 'manual',
-      });
-    } else {
-      this.request();
-    }
-  };
-
-  request = () => {
-    const { api } = this.props;
-    this.setState(
-      {
-        status: 'running',
-      },
-      () => {
-        api.emit(EVENTS.REQUEST, api.getCurrentStoryData().id);
-        // removes all elements from the redux map in store from the previous panel
-        store.dispatch(clearElements());
-      }
-    );
-  };
-
-  render() {
-    const { active } = this.props;
-    if (!active) return null;
-
-    switch (this.state.status) {
-      case 'initial':
-        return <Centered>Initializing...</Centered>;
-      case 'manual':
-        return (
-          <Fragment>
-            <Centered>Manually run the accessibility scan.</Centered>
-            <ActionBar
-              key="actionbar"
-              actionItems={[{ title: 'Run test', onClick: this.request }]}
+  const handleError = useCallback((err: unknown) => {
+    setStatus('error');
+    setError(err);
+  }, []);
+
+  const emit = useChannel({
+    [EVENTS.RESULT]: handleResult,
+    [EVENTS.ERROR]: handleError,
+  });
+
+  const handleManual = useCallback(() => {
+    setStatus('running');
+    emit(EVENTS.MANUAL, storyId);
+  }, [storyId]);
+
+  return (
+    <>
+      {status === 'initial' && <Centered>Initializing...</Centered>}
+      {status === 'manual' && (
+        <>
+          <Centered>Manually run the accessibility scan.</Centered>
+          <ActionBar key="actionbar" actionItems={[{ title: 'Run test', onClick: handleManual }]} />
+        </>
+      )}
+      {status === 'running' && (
+        <Centered>
+          <RotatingIcon inline icon="sync" /> Please wait while the accessibility scan is running
+          ...
+        </Centered>
+      )}
+      {(status === 'ready' || status === 'ran') && (
+        <>
+          <ScrollArea vertical horizontal>
+            <Tabs
+              key="tabs"
+              tabs={[
```

Comment:
this creates a new list every render, we might want to extract into a `useMemo`?